core/src/kmx/kmx_xstring.h

```diff
@@ -8,7 +8,9 @@ namespace kmx {
 
 #define Uni_IsSurrogate1(ch) ((ch) >= 0xD800 && (ch) <= 0xDBFF)
 #define Uni_IsSurrogate2(ch) ((ch) >= 0xDC00 && (ch) <= 0xDFFF)
+
 #define Uni_IsSMP(ch) ((ch) >= 0x10000)
```

Comment:
Misnomer!   SMP is 0x010000…0x01FFFD inclusive.  0x020000 begins plane 2.

Reply:
this macro was used elsewhere, so i didn't 'fix' it

Reply:
Yep, longstanding misnomer. We should rename to `Uni_IsSupplementaryPlane()` which I think covers all planes 1-16?

Reply:
You're right!  https://www.unicode.org/glossary/#supplementary_planes Practically, though, I have not heard this used as much.   Certainly 'the SMP' is only plane 1.  Anyway, I think it depends on what the callers are looking for.

Reply:
Unicode Glossary does not mention SMP. But Supplementary Planes, Supplementary Characters all refer to 1-16. So I blame Unicode for the confusion! Doing a Google search for "supplementary multilingual planes" shows SMP is commonly used to refer to non-BMP. 

This macro was probably introduced around 2002... :grin:

Reply:
"the SMP" - https://www.unicode.org/roadmaps/smp/ this is where i most commonly see it. (UTC)